src/System.IO.Pipelines/src/System/IO/Pipelines/Pipe.cs

```diff
@@ -263,24 +235,18 @@ private void ReturnSegmentUnsynchronized(BufferSegment segment)
 
         internal bool CommitUnsynchronized()
         {
-            if (_writingHead == null)
+            _operationState.EndWrite();
+
+            var bytesWritten = _currentWriteLength;
```

Comment:
It's used in length += bytesWritten. @benaadams @AndyAyersMS does the JIT do good things when you access a field multiple times with in a method. I haven't looked and I'll just assume it does 😄

Reply:
It depends ... if your uses are mostly reads and are reading simple typed fields (not struct fields) and there are no interposing writes then yes it will likely do good things.

If there are interposing writes but they are only and obviously to the same object and the parts of the object you are touching are not weird (overlapping fields or non-field padding areas or fixed array stuff) then it should also do good things.

For more general cases the jit will probably be conservative though.